src/ChromeProfileFormat.jl

```diff
@@ -2,123 +2,284 @@ module ChromeProfileFormat
+    samples    = Int64[] #  Ids of samples leaf.
+    toplevel   = Set{Int64}() #  Ids of samples top nodes.
+    timeDeltas = Int64[] # Time intervals between adjacent samples in microseconds. The first delta is relative to the profile startTime.
+
+    # start decoding backtraces
+    lastwaszero = true
+    current = nothing
+
+    # data is in bottom-up order therefore reverse it
+    for ip in Iterators.reverse(data)
+        # ip == 0x0 is the sentinel value for finishing a backtrace, therefore finising a sample
+        if ip == 0
+            # Avoid creating empty samples
+            if lastwaszero
+                @assert current === nothing
+                continue
             end
+            lastwaszero = true
 
-            JSON.print(f, "scriptId")
-            print(f, ":")
-            JSON.print(f, string(script_id))
-
-            print(f, ",")
-
-            JSON.print(f, "url")
-            print(f, ":")
-            JSON.print(f, v.file)
-
-            print(f, ",")
-
-            # TODO I think this is wrong, because this probably now refers to a line in the function, not the definition of the function?
-            JSON.print(f, "lineNumber")
-            print(f, ":")
-            JSON.print(f, v.line>0 ? v.line-1 : 0)
-
-            print(f, ",")
-
-            # TODO I think this is wrong, because this probably now refers to a line in the function, not the definition of the function?
-            JSON.print(f, "columnNumber")
-            print(f, ":")
-            JSON.print(f, 0)
-
-            print(f, "}")
-
-            print(f, ",")
-
-            JSON.print(f, "children")
-            print(f, ":")
-            JSON.print(f, Int[])
+            if current !== nothing
+                push!(samples, current.id)
+                push!(timeDeltas, period)
+            end
 
-            print(f, "}")
+            current = nothing
+            continue
         end
-        print(f, "]")
-
-        print(f, ",")
+        lastwaszero = false
+
+        # A backtrace consists of a set of IP (Instruction Pointers), each IP points
+        # a single line of code and `litrace` has the necessary information to decode
+        # that IP to a specific frame (or set of frames, if inlining occured).
+
+        # decode the inlining frames
+        for frame in Iterators.reverse(lookup[ip])
+            # ip 0 is reserved
+            frame.pointer == 0 && continue
+            frame.from_c && !from_c && continue
+
+            # A `frame` is an entry in a backtrace. A `ProfileNode` is a method in the call-graph
+            # which can be hit by multiple frames. So first we need to go from frame to method.
+            # As keys we will use `url` + `start_line`.
+            if frame.linfo === nothing # inlined frame
+                file = string(frame.file)
+                name = string(frame.func)
+                start_line = convert(Int64, frame.line) # TODO: Get start_line properly
```

Comment:
@vtjnash This is where my question for getting the `start_line` of a method was motivated from.

Reply:
Means we couldn't compute it. Though if you want it, LLVM gave it to us, we just are choosing not to expose it (https://llvm.org/doxygen/classllvm_1_1DIContext.html)